server/src/main/java/org/elasticsearch/action/fieldcaps/FieldCapabilitiesResponse.java

```diff
@@ -65,6 +65,7 @@ private FieldCapabilitiesResponse(Map<String, Map<String, FieldCapabilities>> re
      */
     FieldCapabilitiesResponse() {
         this.responseMap = Collections.emptyMap();
+        this.indexResponses = Collections.emptyList();
```

Comment:
super minor nit: we say in the javadocs that this is used for serialization, but it's also used when the response is empty (totalNumRequest == 0 in TransportFieldCapabilitiesAction).  Maybe we could have the transport action use one of the other constructors and provide an empty map? And maybe we should validate that what is provided is never null?